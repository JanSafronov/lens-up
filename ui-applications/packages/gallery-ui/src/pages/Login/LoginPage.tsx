import { ChangeEvent, FormEvent, useState } from "react";
import LoginForm from "./components/LoginForm"
import { useLoginToGalleryMutation } from "../../services/GalleryApi";
import { useNavigate } from "react-router-dom";

const LoginPage = () => {
    const navigate = useNavigate();
    const [enterCode, setEnterCode] = useState<string>();

    const [loginToGallery] = useLoginToGalleryMutation();

    const handleFormSubmit = (event: FormEvent) => {
        event.preventDefault()
        // TODO: Add validation
        if (!enterCode) {
            return;
        }

        loginToGallery({ enterCode: enterCode })
            .unwrap()
            .then((payload) => {
                navigate(`/gallery/${payload.galleryId}`)
            })
    }

    const handleEnterCodeInputChange = (event: ChangeEvent<HTMLInputElement>) => {
        // TODO: Validation
        const enterCodeValue = event.target.value;
        setEnterCode(enterCodeValue);
    };

    return (
        <div className="min-h-screen bg-black">
            <div className="container mx-auto flex flex-wrap flex-col md:flex-row justify-center h-screen items-center">
                <LoginForm handleFormSubmit={handleFormSubmit} enterCode={enterCode} handleEnterCodeInputChange={handleEnterCodeInputChange} />
            </div>
        </div>
    )
}

export default LoginPage