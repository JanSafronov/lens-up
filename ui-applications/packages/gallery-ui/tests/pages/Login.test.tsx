import { fireEvent, render } from '@testing-library/react'
import { expect, test } from 'vitest';
import { LoginPage } from '../../src/pages/Login/LoginPage';
import { AppMock } from '../AppMock.tsx';

/**
* @vitest-environment jsdom
*/
test('Entered code must change in the form', async () => {
    AppMock();

    const {asFragment, getByText} = render(
        <LoginPage />,
    );

    const firstRender = asFragment();

    fireEvent.change(getByText("Join to your gallery"));
    
    expect(firstRender).toMatchSnapshot(asFragment());
  });

/**
* @vitest-environment jsdom
*/
test('Form must be handled successfully', () => {
    const {getByText, asFragment} = render(
        <LoginPage />,
    );

    const firstRender = asFragment();

    fireEvent.submit(getByText("Join"));
    
    expect(firstRender).toMatchSnapshot(asFragment());
  });

/**
* @vitest-environment jsdom
*/
test('Login must set gallery successfuly', () => {
  const {getByText, asFragment} = render(
      <LoginPage />,
  );

  const firstRender = asFragment();

  fireEvent.submit(getByText("Join"));
  
  expect(firstRender).toMatchSnapshot(asFragment());
});