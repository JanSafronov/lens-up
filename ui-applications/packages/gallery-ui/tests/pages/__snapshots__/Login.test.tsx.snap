// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Entered code must change in the form 1`] = `{}`;
